app/src/main/java/com/jemiola/moodtimeline/views/timeline/TimelineFragment.kt

```diff
@@ -44,26 +50,78 @@ class TimelineFragment : BaseFragment(), TimelineContract.View {
+            val formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy").withLocale(Locale.ENGLISH)
+            var formatDaty = defaultToDate.format(formatter)
+
+
+            binding.fromEditText.setText(formatDaty)
+            binding.toEditText.setText(formatDaty)
+            setupSearchEditTextColors()
+            setupSearchCalendars()
+            binding.timelineTopLayout.post {
+                initialSearchTopLayoutMoveOutOfScreen()
+                binding.searchImageView.setOnClickListener { onSearchClick() }
+            }
+            setupCalendarFragment()
+            binding.timelineTopLayout.post {
+                initialCalendarTopLayoutMoveOutOfScreen()
+                binding.calendarImageView.setOnClickListener { onCalendarClick() }
+            }
         }
         return binding.root
     }
 
     override fun onStart() {
         super.onStart()
-        counterComeBackLaterInflater = 0
-        presenter.setupTimetableMoods()
+        licznikPowrotuPozniejDoInflacji = 0
+        val callback = createRepositoryCallback<Int>(
+            onSuccessAction = {
+                if (it == 0) showAddEmptyView() else
+                showBottomMenu()
+                val fromDateText = getFromDate()
+                val defaultToDate = repository!!.defaultSearchToDate
+                val formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy").withLocale(Locale.ENGLISH)
+                var formatDaty = defaultToDate.format(formatter)
+                val callback = createRepositoryCallback<List<TimelineMoodBOv2>>(
+                    onSuccessAction = {
+                        val moods = presenter.addSpecialMoodsIfNeeded(it)
+                        setTimelineMoods(moods)
+//                        if (!it.contains(CircleMoodBO.NONE)) {
+                            setupComeBackLaterView()
+//                        }
+                              },
+                    onErrorAction = {}
+                )
+//                repository!!.getTimetableMoods(fromDate, toDate, callback)
+            },
+            onErrorAction = {}
+        )
+        repository!!.getTimetableMoodsCount(callback)
     }
 
-    private fun setupTimeline() {
-        with(binding.timelineRecyclerView) {
-            adapter = TimelineAdapter(this@TimelineFragment)
-            layoutManager = LinearLayoutManager(context)
+    protected fun <T> createRepositoryCallback(onSuccessAction: (result: T) -> Unit, onErrorAction: () -> Unit): OnRepositoryCallback<T> {
+        return object : OnRepositoryCallback<T> {
+            override fun onSuccess(result: T) {
+                onSuccessAction.invoke(result)
+            }
+
+            override fun onError() {
+                onErrorAction.invoke()
+            }
         }
     }
 
+    private fun setupTimeline() {
+
+    }
+
     override fun setTimelineMoods(moods: List<TimelineMoodBOv2>) {
         val adapter = binding.timelineRecyclerView.adapter
         val animationId =
```

Comment:
ale co tutaj przypisujesz? i po co znowu jakieś id, co to znaczy?

- [ ] odpowiedź

Reply:
wczesniej byl null, teraz trzeba cos pobirac